description.md

```diff
@@ -0,0 +1,937 @@
+00000080 00000080 00000080 00000080 00000080 00000080 00000080 00000040
+#                      makerAssetSuffix end offset =>          ^^^^^^^^
+#                      takerAssetSuffix end offset => ^^^^^^^^
+
+# additional parameters for makerAsset function
+# 1st extra parameter (uint256 tokenId) 
+00000000 00000000 00000000 00000000 00000000 00000000 00000000 0000000a
+# 2nd extra parameter (address token)
+00000000 00000000 00000000 5fbdb231 5678afec b367f032 d93f642f 64180aa3
+
+# additional parameters for takerAsset function
+# 1st extra parameter (uint256 tokenId)
+00000000 00000000 00000000 00000000 00000000 00000000 00000000 0000000a
+# 2nd extra parameter (address token)
+00000000 00000000 00000000 e7f1725e 7734ce28 8f8367e1 bb143e90 bb3f0512
+```
+
+### Runtime exchange rate
+
+In some scenarios, the order exchange rate is only known on-chain, for example, if the order implements Dutch auction or the maker wants to get a rate based on an oracle price. To achieve this, the maker can set getter functions to calculate the rate on-chain and then pass them to the `makerAmountGetter` and `takerAmountGetter` extensions. 
+
+These getters are used for:
+
+- `makerAmountGetter`, calculates the maker amount based on the provided taking amount
+- `takerAmountGetter`, calculates the taker amount based on the provided making amount
+
+Both getters should be implemented to retrieve consistent results because it is the taker who decides how the amounts are calculated (based on the making or taking amounts) when the order is executed. See the order fill for details.
+
+Both getters extension calldata have the following structure:
+
+| Address | Selector | Packed arguments |
+| --- | --- | --- |
+| 20 bytes | 4 bytes | variable size calldata |
+
+When the order is executed and the protocol calculates the amount, the selector on the provided contract address is called, and the packed arguments are passed. Additionally the arguments calldata is extended with:
+
+- `requestedAmount` is the requested amount to calculate making or taking amount for.
+- `remainingMakingAmount` is the remaining making amount for the order, which can be different from an order's `makingAmount` if the order was partially filled previously.
+- `orderHash` is the order’s hash.
+
+Therefore, the final call would be as follows:
+
+```solidity
+address.selector(<arguments from calldata>, requestedAmount, remainingMakingAmount, orderHash)
+```
+
+The function is expected to return a `uint256` value that corresponds to the passed `requestedAmount`.
+
+**Example**
+
+The code creates an order that uses `RangeAmountCalculator` to calculate the making and taking amounts.
+
+```javascript
+// Order: 10 weth -> 35000 dai with price range: 3000 -> 4000
+const makingAmount = ether('10');
+const takingAmount = ether('35000');
+const startPrice = ether('3000');
+const endPrice = ether('4000');
+
+const makingAmountGetter = rangeAmountCalculator.address + trim0x(cutLastArg(cutLastArg(
```

Comment:
Should we explain that `rangeAmountCalculator` is object  of deployed contract `RangeAmountCalculator` or is it redundant information?

Reply:
Here the focus is on the calldata produced by the code. A user can implement any code that produces the same calldata. The point is that calldata should be the same as shown below